### ISSUE TITLE
Improve design for OPA - Gatekeeper Config 

### ISSUE THREAD

**Issue by kgroschoff** [kubermatic/dashboard#3078]
The Gatekeeper Config needs some design improvements.

Things that might be relevant:
- This tab is only displayed, if OPA is enabled.
- Initially it is empty, but also this whole config is optional.
- There will be only exactly one config at a time.
- If there is no config, display `Add` button.
- If there is a config, display `Edit` and `Delete` buttons; hide `Add`. 
- For now the config is just a spec (see image below), this _might_ change in the future but not sure yet.

Currently it is looking like this:
![gkcs](https://user-images.githubusercontent.com/19547196/108077552-ee328800-706c-11eb-896a-52bad08189f2.PNG)
![gkc](https://user-images.githubusercontent.com/19547196/108077557-ef63b500-706c-11eb-98a9-0be2ad6f63de.PNG)

**Comment by kgroschoff** [comment]
cc @cschieder

**Comment by maciaszczykm** [comment]
In my opinion we should avoid using titles twice. I think `Gatekeeper Config` should be present only in the tab header.

**Comment by kgroschoff** [comment]
Agree. But actually I just inserted the wrong screenshots; title isn't merged. Good catch ;)

**Comment by cschieder** [comment]
About the config part, is it necessary that we show it as code? As an alternative we could pick the essential information and show it like we show data on machine deployments/clusters.

**Comment by kgroschoff** [comment]
Would love that, but not sure if it is possible. 🤔 
@lsviben any idea/thoughts/comments? 😅

**Comment by lsviben** [comment]
@

> Would love that, but not sure if it is possible. 
> @lsviben any idea/thoughts/comments?

It is, the fields for Config are well defined so you can show them more nicely. Also I think it would be great to have it for add and edit dialogue, instead of just code, using fields. AFAIK the code stuff is just a solution for the beginning to get things rolling.

**Comment by cschieder** [comment]
what do you think of this? 
![image](https://user-images.githubusercontent.com/49145827/108972282-6c9db400-7683-11eb-8d09-dc078f890d1a.png)

**Comment by floreks** [comment]
How about moving the `Readiness` entry to the top to be in line with edit/delete buttons. Right now it feels like there is a lot of space wasted on the top of the card, especially on the wider screen.

I'd also like to see how aligning titles `Sync|Match|Validation` with the table instead of column name would look like. It feels a bit artificial and enforced, not that natural.

**Comment by maciaszczykm** [comment]
> How about moving the Readiness entry to the top to be in line with edit/delete buttons. Right now it feels like there is a lot of space wasted on the top of the card, especially on the wider screen.

+1

**Comment by cschieder** [comment]
@floreks @maciaszczykm 
![image](https://user-images.githubusercontent.com/49145827/109287895-7a3e6f80-7824-11eb-963c-8d4a2b724238.png)
